### PR TITLE
Improve h3d layout methods

### PR DESCRIPTION
**Pull request by maxcapodi78**
add few methods to hfss3dlayout.py:
- dissolve_component
- create_pec_on_component_by_nets
- modeler.create_components_on_pins
Added new property to set_solderball method to add an offset for the port in component solderball.
added method create_components_on_pins

### CODE REVIEW

**Review by Samuelopez-ansys — DISMISSED** [review]
LGTM

File: pyaedt/hfss3dlayout.py [hunk 1]
```diff
@@ -363,6 +363,24 @@ def create_wave_port_from_two_conductors(self, primivitivenames=[""], edgenumber
...
             return False
 
+    @pyaedt_function_handler()
+    def dissolve_component(self, component_name):
+        """Dissolve a component and remove it from 3D Layout.
+
+        Parameters
+        ----------
+        component_name : str
+            Component Name
```
Comment (MaxJPRey) [inline pyaedt/hfss3dlayout.py:373]:
```suggestion
            Name of the components.
```

File: pyaedt/hfss3dlayout.py [hunk 2]
```diff
@@ -363,6 +363,24 @@ def create_wave_port_from_two_conductors(self, primivitivenames=[""], edgenumber
...
             return False
 
+    @pyaedt_function_handler()
+    def dissolve_component(self, component_name):
+        """Dissolve a component and remove it from 3D Layout.
+
+        Parameters
+        ----------
+        component_name : str
+            Name of the components.
```
Comment (PipKat) [inline pyaedt/hfss3dlayout.py:373]: Both the method description and the parameter name imply a singular component. I think the parameter description should also use the singular form.

File: pyaedt/hfss3dlayout.py [hunk 3]
```diff
@@ -363,6 +363,24 @@ def create_wave_port_from_two_conductors(self, primivitivenames=[""], edgenumber
...
+
+        Returns
+        -------
+        bool
```
Comment (PipKat) [inline pyaedt/hfss3dlayout.py:377]:
```suggestion
        bool
            ``True`` when successful, ``False`` when failed.
```

File: pyaedt/hfss3dlayout.py [hunk 4]
```diff
@@ -407,6 +425,39 @@ def create_ports_on_component_by_nets(
...
+            pass
+        else:
+            nets = [nets]
+        net_array = ["NAME:Nets"] + nets
+        self.oeditor.CreatePortsOnComponentsByNet(["NAME:Components", component_name], net_array, "PEC", "0", "0", "0")
+        return True
+
     @pyaedt_function_handler()
     def create_differential_port(self, via_signal, via_reference, port_name, deembed=True):
         """Create a new differential port.
```
Comment (PipKat) [inline pyaedt/hfss3dlayout.py:463]:
```suggestion
        """Create a differential port.
```
Create means "new" so it's not necessary.

File: pyaedt/modeler/pcb/Primitives3DLayout.py [hunk 5]
```diff
@@ -1489,6 +1489,59 @@ def place_3d_component(
...
 
+    @pyaedt_function_handler()
+    def create_components_on_pins(
+        self,
+        pins,
+        definition_name=None,
+        component_type="Other",
+        ref_des="U100",
+    ):
+        """Create a new component based on a pin list.
```
Comment (PipKat) [inline pyaedt/modeler/pcb/Primitives3DLayout.py:1500]:
```suggestion
        """Create a component based on a pin list.
```

File: pyaedt/hfss3dlayout.py (hunk 2 above)
Comment (maxcapodi78) [inline pyaedt/hfss3dlayout.py:373]:
```suggestion
            Name of the component.
```

**Pull request merged**